### PR TITLE
Bye bye testify mock 2

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: crypto/trillian_signer_test.go [hunk 1]
```diff
@@ -4,78 +4,73 @@ import (
...
+	if !ok {
+		return false
+	}
+	return h.HashFunc() == crypto.SHA256
 }
-
-func messageHash() []byte {
-	h := trillian.NewSHA256()
-	return h.Digest([]byte(message))
+func (i usesSHA256Hasher) String() string {
```
Comment (Martin2112) [inline crypto/trillian_signer_test.go]: Missing blank line?

Comment (AlCutter) [inline crypto/trillian_signer_test.go]: added

File: merkle/sparse_merkle_tree_test.go [hunk 2]
```diff
@@ -126,58 +139,73 @@ func TestRootAtRevision(t *testing.T) {
...
+
+	r, tx := getSparseMerkleTreeReaderWithMockTX(mockCtrl, 100)
 	n1, n2 := getRandomRootNode(t, 14), getRandomRootNode(t, 15)
-	tx.On("Commit").Return(nil)
-	tx.On("GetMerkleNodes", int64(23), mock.MatchedBy(isRootNodeOnly)).Return([]storage.Node{n1, n2}, nil)
+	tx.EXPECT().Commit().AnyTimes().Return(nil)
+	tx.EXPECT().GetMerkleNodes(int64(23), rootNodeMatcher{}).Return([]storage.Node{n1, n2}, nil)
 	_, err := r.RootAtRevision(23)
 	if err == nil || err == ErrNoSuchRevision {
 		t.Fatalf("Attempt to retrieve root an non-existent revision did not result in error: %v", err)
```
Comment (Martin2112) [inline merkle/sparse_merkle_tree_test.go]: It looks like I might have cut and pasted this message from the test above. It should say that it retrieved the wrong root for the requested revision.

Comment (Martin2112) [inline merkle/sparse_merkle_tree_test.go]: Wait, that's a sparse tree test, not the log - I'll just step quietly backwards into this hedge.

Comment (AlCutter) [inline merkle/sparse_merkle_tree_test.go]: hehe

**Comment by Martin2112** [comment]
LGTM when build fixed.

**Pull request merged**